app/database/operator/utils/reaction.ts

```diff
@@ -26,18 +26,21 @@ export const sanitizeReactions = async ({database, post_id, rawReactions, skipSy
         fetch());
 
     // similarObjects: Contains objects that are in both the RawReaction array and in the Reaction table
-    const similarObjects: ReactionModel[] = [];
+    const similarObjects = new Set<ReactionModel>();
 
     const createReactions: RecordPair[] = [];
 
+    const reactionsMap = reactions.reduce((result, reaction) => {
+        result[`${reaction.userId}-${reaction.emojiName}`] = reaction;
+        return result;
+    }, {} as Record<string, ReactionModel>);
```

Comment:
Nit: Instead of `as`, you can define the generic (i.e. `reduce<Recrod<string, ReactionModel>>`).